frontends/web/src/routes/account/info/buyCTA.tsx

```diff
@@ -0,0 +1,30 @@
+import { h, FunctionalComponent } from 'preact';
+import { route } from 'preact-router';
+import { IBalance } from '../../../api/account';
+import { Button } from '../../../components/forms';
+import { translate, TranslateProps } from '../../../decorators/translate';
+import * as styles from './buyCTA.module.css';
+
+interface BuyCTAProps {
+    code: string;
+    balance?: IBalance
+}
+
+type Props = BuyCTAProps & TranslateProps;
+
+const BuyCTA :FunctionalComponent<Props>  = ({code, balance, t}) => {
```

Comment:
Cool, i don't have a personal defined way of what is better, still there is one tradeoff that is not listed, meanwhile you are defining the type of the input(`Props` or `PropsWithChildren<Props>`), and the output (`JSX.Element` or `ReactElement<any, any> | null`), using the `FunctionComponent` defines a more generic way and makes it more integral, you don't need to import or know exact types of the rendering lib you are using. 

For example this is the definition of FunctionalComponent on preact:
```typescript
interface FunctionalComponent<P = {}> {
  (props: RenderableProps<P>, context?: any): VNode<any> | null;
  displayName?: string;
  defaultProps?: Partial<P>;
}
```
and this is on React
```typescript
interface FunctionComponent<P = {}> {
  (props: PropsWithChildren<P>, context?: any): ReactElement<any, any> | null;
  propTypes?: WeakValidationMap<P> | undefined;
  contextTypes?: ValidationMap<any> | undefined;
  defaultProps?: Partial<P> | undefined;
  displayName?: string | undefined;
}
```

the specifics typings of every engine for using with more internal tools like `React.Children API` are defined by default in each `FunctionComponent` typing, and switching between libs os more difficult if every component is defined strictly to their input/output. Of course this is not a big problem as it seems because any function declared as `(props):Props => JSX.Element` would be a FC in any lib.

This lead me to think that if we are going to migrate to React, we will need to migrate a lot of `RenderedProps<P>` to `PropsWithChildren<P>`. Wouldnt be easier to have a typing definition that abstract this difference, example using `WithProps<P> = RenderedProps<P>`, and so when we swithc to React we only have to change one file?

Still, if the team decides to use `(props):Props => JSX.Element` as default typing for components im okay with that, as more integral decisions are made its easier.